lib/function/matrix/chol.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+module.exports = function (math) {
+  var Matrix = require('../../type/Matrix'),
+    collection = require('../../type/collection'),
+
+    isCollection = collection.isCollection;
+  /**
+   * Cholesky factorization math.chol(x, p). 
+   *
+   * The second parameter p is optional. If not provided, it defaults to 'upper'.
+   *
+   *  math.chol(A) produces an upper triangular R so that R'*R = A.
+        If A is not positive definite, complex element is returned.
+ 
+      math.chol(A, 'lower') produces a lower triangular L so that L*L' = A.
+        If A is not positive definite, complex element is returned.
+        
+   * Reference: http://www.math.sjsu.edu/~foster/m143m/cholesky.pdf
+   *
+   * Syntax:
+   *
+   *    math.chol(x)
+   *    math.chol(x, p)
+   *
+   * Examples:
+   *
+   *    var A = [[3, 3], [5, 8]];               // returns [[3, 3], [5, 8]]
+   *    math.chol(A, 'lower');                  // returns [[ 1.7320508075688772, 0 ],
+                                                //          [ 1.7320508075688776, 2.2360679774997894 ]]
+   *
+   *
+   * See also:
+   *
+   *    inv
+   *
+   * @param  {Array | Matrix} x Array | Matrix to factor.
+   * @param  {String} [p='upper'] 'upper' || 'lower', default is 'upper'
+   * @return {Array | Matrix} the triangular matrix
+   */
+  math.chol = function chol(x, p) {
+    var array = (x instanceof Matrix) ? x.valueOf() : x;
```

Comment:
This implementation will not have a good performance on Sparse Matrices. Calling `matrix.valueOf()` is very expensive on those types of matrices. Implementation should delegate the specifics to the actual Matrix type (Dense, CCS, CRS).